toolchain/base/in_flight_clang.h

```diff
@@ -0,0 +1,62 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_BASE_IN_FLIGHT_CLANG_H
+#define CARBON_TOOLCHAIN_BASE_IN_FLIGHT_CLANG_H
+
+#include <thread>
+
+#include "clang/AST/ASTContext.h"
+#include "clang/Basic/Diagnostic.h"
+#include "clang/Basic/SourceManager.h"
+#include "clang/CodeGen/CodeGenAction.h"
+#include "clang/CodeGen/ModuleBuilder.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/IntrusiveRefCntPtr.h"
+
+namespace Carbon {
+class InFlightClang {
```

Comment:
Seems like this really should have some detailed class comments to explain how it works and what it does?